### PR TITLE
use data instead of i32.store to store storage keys in memory

### PR DESCRIPTION
**Pull request by hugo-dc**

### CODE REVIEW

File: src/GeneralStateTestsFiller/stEWASMTests/callNoDataNoValueFiller.yml [hunk 1]
```diff
@@ -105,7 +104,6 @@ callNoDataNoValue:
             (set_local $memStart (i32.const 32))
 
             ;; init pointers
-            (set_local $ptrStorageKey1 (get_local $memStart))
```
Comment (axic) [inline src/GeneralStateTestsFiller/stEWASMTests/callNoDataNoValueFiller.yml]: I think in this case you weren't supposed to remove this but the `i32.store` line a few lines down?

File: src/GeneralStateTestsFiller/stEWASMTests/codeCopyFullyOOBFiller.yml [hunk 2]
```diff
@@ -21,6 +21,12 @@ codeCopyFullyOOB:
           (import "ethereum" "getCodeSize" (func $getCodeSize (result i32)))
           (import "ethereum" "storageStore" (func $storageStore (param i32 i32)))
           (memory 1)
+          ;; $ptrStorageKey1
+          (data (i32.const 0)  "\01\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 1
+          ;; $ptrStorageKey2
+          (data (i32.const 32) "\02\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 2
+          ;; ptrStorageVal2 - arbitrary value, just make sure it is not written to storage
+          (data (i32.const 64) "\01\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 1
```
Comment (axic) [inline src/GeneralStateTestsFiller/stEWASMTests/codeCopyFullyOOBFiller.yml]: In the previous code this was at offset 96.

File: src/GeneralStateTestsFiller/stEWASMTests/codeCopyPartiallyOOBFiller.yml [hunk 3]
```diff
@@ -21,6 +21,9 @@ codeCopyPartiallyOOB:
           (import "ethereum" "getCodeSize" (func $getCodeSize (result i32)))
           (import "ethereum" "storageStore" (func $storageStore (param i32 i32)))
           (memory 1)
+          ;; init memory
+          (data (i32.const 0) "\01\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 1
+          (data (i32.const 1) "\02\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 2
```
Comment (axic) [inline src/GeneralStateTestsFiller/stEWASMTests/codeCopyPartiallyOOBFiller.yml]: I think this should be `const 32`

File: src/GeneralStateTestsFiller/stEWASMTests/extCodeCopyFullyOOBFiller.yml [hunk 4]
```diff
@@ -37,6 +37,9 @@ extCodeCopyFullyOOB:
           (import "ethereum" "storageStore" (func $storageStore (param i32 i32)))
           (memory 1)
           (data (i32.const 0) "\01\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\ef\be\ad\de")
+          ;; storage keys
+          (data (i32.const 32) "\01\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 1
+          (data (i32.const 96) "\02\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 2
           (export "memory" (memory 0))
           (export "main" (func $main))
```
Comment (axic) [inline src/GeneralStateTestsFiller/stEWASMTests/extCodeCopyFullyOOBFiller.yml:45]: This didn't remove the `i32.store`s for `ptrStorageKey*`.

File: src/GeneralStateTestsFiller/stEWASMTests/revertFiller.yml [hunk 5]
```diff
@@ -68,6 +66,10 @@ revert:
           (memory 1)
           (data (i32.const 0)  "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\ef\be\ad\de")
 
+          ;; storage keys
+          (data (i32.const 128) "\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 0
+          (data (i32.const 160) "\01\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 1
+          (data (i32.const 192) "\02\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00\00") ;; 2
           (export "memory" (memory 0))
           (export "main" (func $main))
```
Comment (axic) [inline src/GeneralStateTestsFiller/stEWASMTests/revertFiller.yml:75]: The `i32.store`s aren't removed in this file for those locations.

**Comment by hugo-dc** [comment]
@axic thanks for reviewing, fixed according to your comments.

**Review by axic — APPROVED** [review]

**Pull request merged**